python/plugins/processing/algs/qgis/Rasterize.py

```diff
@@ -0,0 +1,282 @@
+
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+"""
+
+from processing.core.outputs import OutputRaster
+
+from processing.algs.qgis.QgisAlgorithm import QgisAlgorithm
+
+from qgis.PyQt.QtGui import QImage, QPainter
+from qgis.PyQt.QtCore import QSize
+from qgis.core import (
+    QgsMapSettings,
+    QgsMapRendererCustomPainterJob,
+    QgsRectangle,
+    QgsProject,
+    QgsProcessingException,
+    QgsProcessingParameterExtent,
+    QgsProcessingParameterString,
+    QgsProcessingParameterNumber,
+    QgsProcessingParameterRasterLayer,
+    QgsProcessingOutputRasterLayer,
+    QgsProcessingParameterRasterDestination
+)
+
+import qgis
+import osgeo.gdal
+import os
+import tempfile
+import math
+
+__author__ = 'Matthias Kuhn'
+__date__ = '2016-10-05'
+__copyright__ = '(C) 2016 by OPENGIS.ch'
+
+# This will get replaced with a git SHA1 when you do a git archive
+
+__revision__ = '$Format:%H$'
+
+
+class RasterizeAlgorithm(QgisAlgorithm):
+    """Processing algorithm renders map canvas to a raster file.
+    It's possible to choose the following parameters:
+        - Map theme to render
+        - Layer to render
+        - The minimum extent to render
+        - The tile size
+        - Map unit per pixel
+        - The output (can be saved to a file or to a temporary file and
+          automatically opened as layer in qgis)
+    """
+
+    # Constants used to refer to parameters and outputs. They will be
+    # used when calling the algorithm from another algorithm, or when
+    # calling from the QGIS console.
+
+    OUTPUT_LAYER = 'OUTPUT_LAYER'
+    MAP_THEME = 'MAP_THEME'
+    LAYER = 'LAYER'
+    EXTENT = 'EXTENT'
+    TILE_SIZE = 'TILE_SIZE'
+    MAP_UNITS_PER_PIXEL = 'MAP_UNITS_PER_PIXEL'
+
+    def __init__(self):
+        super().__init__()
+
+    def initAlgorithm(self, config=None):
+        """Here we define the inputs and output of the algorithm, along
+        with some other properties.
+        """
+
+        # The parameters
+        self.addParameter(
+            QgsProcessingParameterString(self.MAP_THEME,
+                                         description=self.tr(
+                                             'Map theme to render.'),
+                                         defaultValue=None, optional=True))
+
+        self.addParameter(
+            QgsProcessingParameterRasterLayer(self.LAYER, description=self.tr(
```

Comment:
Why is this restricted to raster layers only? I think QgsProcessingParameterMapLayer makes more sense here so that users can choose to render single vector layers if desired.